build.gradle

```diff
@@ -32,6 +32,11 @@ allprojects {
         ignoreExitValue true
         systemProperty 'BASELOGDIR', project.baseLogDir
         systemProperty 'BASEREFLECTIONDIR', project.baseReflectionDir
+        workingDir = rootProject.projectDir
+        def sourceDirs = sourceSets.main.java.srcDirs[0].absolutePath
+        def projectDir = project.rootDir.absolutePath
+        def relativePath = new File(projectDir).toURI().relativize(new File(sourceDirs).toURI()).getPath()
```

Comment:
java.io.File ist abgelöst durch java.nio.Path. vermutlich reicht dir hier sowas wie `Path.of(projectDir).relativize(Path.of(sourceDir))`???